src/Core/Module/ModuleRepository.php

```diff
@@ -282,7 +282,7 @@ private function getModulesFromHook()
         if ($this->modulesFromHook === null) {
             $modulesFromHook = $this->hookManager->exec('actionListModules', [], null, true);
             $modulesFromHook = array_values($modulesFromHook ?? []);
-            $this->modulesFromHook = empty(reset($modulesFromHook)) ? [] : array_merge(...$modulesFromHook);
+            $this->modulesFromHook = empty($modulesFromHook) ? [] : array_merge(...$modulesFromHook);
```

Comment:
I think we need to check both then:
```suggestion
            $this->modulesFromHook = empty($modulesFromHook) || empty(reset($modulesFromHook)) ? [] : array_merge(...$modulesFromHook);
```

Reply:
If we do this, if the first array is empty, we will take an empty modules list too

Reply:
I think we don't need the "if empty"

Doing this will be just fine

```
$modulesFromHook = array_values($modulesFromHook ?? []);
$this->modulesFromHook = array_merge(...$modulesFromHook);
```

Reply:
I'm afraid the spread operator in arrays is since 7.4, am I right?

Reply:
I make it more simple with foreach and array merge